src/model/ai/evaluation/GameOverEvaluator.java

```diff
@@ -0,0 +1,33 @@
+package model.ai.evaluation;
+
+import model.Board;
+import model.PieceColor;
+
+public class GameOverEvaluator extends BoardEvaluator {
+
+    public GameOverEvaluator() {
+        super();
+    }
+
+    public GameOverEvaluator(double weight) {
+        super(weight);
+    }
+
+    @Override
+    public double evaluateBoard(Board theBoard, PieceColor color) {
+        double value = 0.0;
+
+        PieceColor winner = theBoard.determineWinner();
+
+        if (winner == null) {
```

Comment:
This is a case where returning null makes sense for the given implementation of `determineWinner()`, rather than having an `IllegalStateException` be thrown. That being said, perhaps we use the Null Object pattern again and create a NullColor? This could also be the color used for the NullPiece object. Just thinking out loud.